### ISSUE TITLE
where is the code of freezing the blocks that you don't want to finetune?

### ISSUE THREAD

**Issue by A-zhudong** [pengzhiliang/MAE-pytorch#70]
Hello, thanks for your implementation.

I have read the main part of your code, but I didn't find the code that controls the Partial fine-tuning. Could you please tell me where is that part in "run_class_finetuning.py", "modeling_finetune.py" or anywhere else?

Wainting for your reply, thank you.

**Comment by SUNJIMENG** [comment]
@A-zhudong I have the same question as you. 
Waiting for reply.

**Comment by SoonFa** [comment]
I have the same question as you.
Waiting for reply.

**Comment by pengzhiliang** [comment]
Hello, sorry for the late reply!

As mentioned in BEiT and MAE, in the end to end fintuning procedure, No blocks/layers need beed frozen.
But the LR of each blocks in different, you can find it in [here](https://github.com/pengzhiliang/MAE-pytorch/blob/8b3ab30f99000ac56b2469de6fd48341ed799a21/optim_factory.py?_pjax=%23js-repo-pjax-container%2C%20div%5Bitemtype%3D%22http%3A%2F%2Fschema.org%2FSoftwareSourceCode%22%5D%20main%2C%20%5Bdata-pjax-container%5D#L45)

**Comment by A-zhudong** [comment]
> Hello, sorry for the late reply!
> 
> As mentioned in BEiT and MAE, in the end to end fintuning procedure, No blocks/layers need beed frozen. But the LR of each blocks in different, you can find it in [here](https://github.com/pengzhiliang/MAE-pytorch/blob/8b3ab30f99000ac56b2469de6fd48341ed799a21/optim_factory.py?_pjax=%23js-repo-pjax-container%2C%20div%5Bitemtype%3D%22http%3A%2F%2Fschema.org%2FSoftwareSourceCode%22%5D%20main%2C%20%5Bdata-pjax-container%5D#L45)

Thanks for your reply.

But it seems that they did frozen some blocks and tested the effect as mentioned in "Masked Autoencoders Are Scalable Vision Learners". 

Maybe it is better if we implement that part? 
![1640162609(1)](https://user-images.githubusercontent.com/41557393/147063254-e9b51336-52e3-47e8-80c8-9c7bb09b149c.png)

**Comment by pengzhiliang** [comment]
Oh, I am sorry that I forget this.

You just need to freeze the blocks you want to, in the __init__ function.

**Comment by A-zhudong** [comment]
OK, thank you.
